src/main/java/seedu/address/model/person/Person.java

```diff
@@ -44,6 +45,40 @@ public Person(Name name, Phone phone, Optional<Email> email, Optional<Address> a
         this.dateOfLastVisit = dateOfLastVisit;
         this.emergencyContact = emergencyContact;
         this.remark = remark;
+        this.hasFullViewToggled = false;
+    }
+
+    /**
+     * Constructor for creating an edited person.
+     */
+    public Person(Name name, Phone phone, Optional<Email> email, Optional<Address> address, Set<Tag> tags,
+                  Optional<DateOfLastVisit> dateOfLastVisit, Optional<EmergencyContact> emergencyContact,
+                  Remark remark, boolean hasFullViewToggled) {
+        requireAllNonNull(name, phone, email, address, tags, dateOfLastVisit, remark);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.address = address;
+        this.tags.addAll(tags);
+        this.dateOfLastVisit = dateOfLastVisit;
+        this.emergencyContact = emergencyContact;
+        this.remark = remark;
+        this.hasFullViewToggled = hasFullViewToggled;
+    }
+
+    /**
+     * Constructs an identical Person with view toggled.
+     */
+    private Person(Person toHaveViewToggled) {
```

Comment:
Since this is a boolean personToView doesn't work well but how about `isBeingViewed`?

Reply:
sorry this might be some misunderstanding on my part but I was referring to line 72 `Person(Person toHaveViewToggled)` and it is a `Person` object not a `boolean`?